src/lambda/api/providerDownload.go

```diff
@@ -65,6 +66,10 @@ func downloadProviderVersion(config config.Config) LambdaFunc {
 		if err != nil {
 			// log the error too for dev
 			slog.Error("Error getting version", "error", err)
+			var ghErr *github.ProviderError
+			if errors.As(err, &ghErr) {
+				return events.APIGatewayProxyResponse{StatusCode: ghErr.Code}, err
```

Comment:
```suggestion
				return events.APIGatewayProxyResponse{StatusCode: ghErr.Code}, nil
```
as we already handled that error, we don't want to keep returning it.
this makes the lambda to keep returning 502